### ISSUE TITLE
[Bug]: White scrollbar with dark themes

### ISSUE THREAD

**Issue by IronZDev** [carbon-design-system/carbon#15594]
### Package

@carbon/react

### Browser

Chrome, Safari

### Package version

1.48.1

### React version

18.2.0

### Description

The scrollbar is not colored according to the dark themes (g90, g100).
A workaround is to manually set the `color-scheme` CSS property.
The issue does not occur on Firefox.
| Issue  | How it should look like |
| ------------- | ------------- |
| ![image](https://github.com/carbon-design-system/carbon/assets/39484495/a4443352-cf1e-4f4a-bc1c-18b45886abcb)  | ![image](https://github.com/carbon-design-system/carbon/assets/39484495/458af069-53d2-42ed-8ee9-e6340aab0958) |

### Reproduction/example

https://stackblitz.com/edit/github-syzh29?file=src%2FApp.jsx

### Steps to reproduce

1. Switch to g90/g100 theme.
2. Create any component with a scrollbar.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Thanks for reporting this, I think it's definitely worth exploring. My concerns and questions include:

* Does defining this property override values we have set (background, borders, etc.) for the components in the g90, g100 themes?
* Is there a way we could scope this setting to just impact the scrollbars?
* Applying this globally seems reasonable, but will it also work with scoped/inline/zoned theming where only a portion of the page is in a different theme?
* Should this be configurable, or built into the themes? e.g. - would we need to document when a user would need to override this setting? (inline/zoned theming, maybe?)
* Would a better approach be to wait for [`scrollbar-color`](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color) support so we can explicitly define scrollbar colors to meet 3:1 color ratio for each of our themes?
* How are other design systems approaching this? (spectrum, fluent, chakra, radix, etc)
  * Radix [ScrollArea](https://www.radix-ui.com/primitives/docs/components/scroll-area)

Curious if you have any thoughts on these - if not, I'd love to hear other thoughts or opinions from more people in the community.

This partially relates to https://github.com/carbon-design-system/gatsby-theme-carbon/issues/1399

**Comment by IronZDev** [comment]
While I do not currently have answers for all these questions, I can answer some: 

- Is there a way we could scope this setting to just impact the scrollbars?

To my best knowledge, I am afraid we cannot. However, I encourage the community to confirm this. 
- Applying this globally seems reasonable, but will it also work with scoped/inline/zoned theming where only a portion of the page is in a different theme?

Yes, we would have to override the `color-scheme` when we change the theme somewhere with `Theme` component or any other technique. Example:
https://stackblitz.com/edit/github-syzh29-q3bcpl?file=src%2FApp.jsx
- Should this be configurable, or built into the themes? e.g. - would we need to document when a user would need to override this setting? (inline/zoned theming, maybe?)

I believe it should be built into themes. 
- Would a better approach be to wait for [scrollbar-color](https://developer.mozilla.org/en-US/docs/Web/CSS/scrollbar-color) support so we can explicitly define scrollbar colors to meet 3:1 color ratio for each of our themes?

We might also use [webkit-scrollbar](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar), which does exactly that and is already supported in Chrome and Safari, where the problem occurs.

- How are other design systems approaching this? (spectrum, fluent, chakra, radix, etc)
So far, I have only checked the ChakraUI, and it sets the color-scheme on root HTML element when setting the Theme. Example:
https://codesandbox.io/p/sandbox/exciting-yonath-72znj4

**Comment by mariuszmickiewicz** [comment]
Adding `color-scheme: dark;` to `:root` for dark themes at Carbon level seems like a good solution to me. My assumption is that it would work well with components which specifically have their theme fixed (comparing to the theme of the whole page) based on the fact that each theme loads their own CSS variables.

> User agents change the following aspects of the UI
> The color of the canvas surface

Not sure if any of Carbon components uses it.

> The default colors of scrollbars and other interaction UI. 

Clear benefit - currently dark theme scrollbars look broken.

> The default colors of form controls.

Except from Select component (which would also benefit from matching the theme of the page) all other form elements are styled I believe.

> The default colors of other browser-provided UI, such as "spellcheck" underlines. 

That is also a benefit and without that probably we have an accessibility issue.


<img width="367" height="677" alt="Image" src="https://github.com/user-attachments/assets/0482c6b8-a729-4290-ac2b-b307e7059825" />

<img width="386" height="681" alt="Image" src="https://github.com/user-attachments/assets/07deac93-2c05-4198-9101-f03ded58eae5" />
